hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/StreamFileProducerSingleThreaded.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.records.impl.producers;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.hedera.hapi.node.state.blockrecords.RunningHashes;
+import com.hedera.hapi.streams.HashAlgorithm;
+import com.hedera.hapi.streams.HashObject;
+import com.hedera.node.app.records.impl.BlockRecordStreamProducer;
+import com.hedera.node.app.spi.info.SelfNodeInfo;
+import com.hedera.node.app.workflows.handle.record.SingleTransactionRecord;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A single threaded implementation of {@link BlockRecordStreamProducer} where all operations happen in a blocking
```

Comment:
Originally, this implementation was just in the test package, and was used as a way to verify that the complex multi-threaded implementation was correct, by verifying that the tests worked with this one, and if they failed with the other, then it indicated an error in the concurrent implementation. I then moved this into the actual sources, because we're going to want to use this one with the "hedera local node". Since we had the implementation anyway, I thought it best to bring it over so we can use it. I also thought it would be good to have so we could do some A/B performance testing.